build.gradle.kts

```diff
@@ -5,12 +5,14 @@ plugins {
     kotlin("jvm") apply false
     id("java")
     id("org.jetbrains.dokka") version "1.7.20"
-    id("io.github.gradle-nexus.publish-plugin")
+    id("io.github.gradle-nexus.publish-plugin") version "1.1.0"
+    id("build-logic") apply false
```

Comment:
Is there any specific reason that `build-logic` is not applied here, but applied within `allprojects {}`? Makes me think it'll break something if I remove `apply false` here and `apply(plugin = "build-logic")` down below, but nothing bad seems to happen

Reply:
If so, we should apply it in `plugins` and `subprojects`, to avoid redundant applying.

Actually, we should just apply `build-logic` in where needed instead of all projects